x-pack/plugins/infra/public/pages/logs/log_entry_rate/page_providers.tsx

```diff
@@ -29,16 +31,19 @@ export const LogEntryRatePageProviders: React.FunctionComponent = ({ children })
 
   const { space } = useActiveKibanaSpace();
 
+  const { idFormats, isLoadingLogAnalysisIdFormats, hasFailedLoadingLogAnalysisIdFormats } =
+    useLogMlJobIdFormatsShimContext();
+
   // This is a rather crude way of guarding the dependent providers against
   // arguments that are only made available asynchronously. Ideally, we'd use
   // React concurrent mode and Suspense in order to handle that more gracefully.
   if (space == null) {
     return null;
   } else if (!isPersistedLogView) {
     return <InlineLogViewSplashPage revertToDefaultLogView={revertToDefaultLogView} />;
-  } else if (isLoading || isUninitialized) {
+  } else if (isLoading || isUninitialized || isLoadingLogAnalysisIdFormats) {
```

Comment:
We could remove the `idFormats!` below by doing this: 

```suggestion
  } else if (isLoading || isUninitialized || isLoadingLogAnalysisIdFormats || !idFormats) {
```